### DISCUSSION TITLE
关于wpf使用mvvm模式通过绑定隐藏窗口的问题

### DISCUSSION THREAD

**Discussion by 158545614** [BYJRK/DotNet-Discussions#383] [MVVM]
我有一个window  ，window 的visibility 绑定到viewmodel的一个bool属性（名字叫做windowVisibility），并且通过convert转换成visibility ，但是我在viewmodel里把windowVisibility设置为false，window不会隐藏。是不是window 的visibility没法通过这种方式绑定，还是其他问题，求各位指点一下

代码如下
<Window xmlns:uctl="clr-namespace:UserControlLibrary.RingProgressCtl;assembly=UserControlLibrary"  x:Class="RingAnimationProject.RingAnimation"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:local="clr-namespace:RingAnimationProject"
        mc:Ignorable="d"
        d:DataContext="{d:DesignInstance local:RingAnimationViewModel, IsDesignTimeCreatable=True}"
        Title="RingAnimation"  Height="80" Width="80" AllowsTransparency="True" WindowStyle="None" Background="Transparent" WindowStartupLocation="CenterScreen"
        Visibility="{Binding WindowVisibility,Converter= {local:BoolToVisibilityConvert}}">
    <Grid>
    </Grid>
</Window>

public partial class RingAnimation : Window
{

    public RingAnimation()
    {
        InitializeComponent();
        this.Topmost = true;
    }
}

public partial class RingAnimationViewModel:ObservableRecipient
{
    [ObservableProperty]
    private bool windowVisibility;
}

public class BoolToVisibilityConvert : MarkupExtension, IValueConverter
{

    public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
    {
        if ((bool)value)
        {
            return Visibility.Visible;
        }
        else
        {
            return Visibility.Hidden;
        }
    }
    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
    {
        Visibility visibility = (Visibility)value;
        if (visibility == Visibility.Visible)
        {
            return true;
        }
        else
        {
            return false;
        }
    }

    public override object ProvideValue(IServiceProvider serviceProvider)
    {
        return this;
    }

}

**Comment by 158545614** [comment]
datacontent是通过依赖注入
private static IServiceProvider ConfigServices()
{
    ServiceCollection services = new ServiceCollection();
    services.AddSingleton<RingAnimationViewModel>();
    services.AddSingleton(sp=>new RingAnimation { DataContext = sp.GetService<RingAnimationViewModel>() });
    return services.BuildServiceProvider();

}

**Comment by arcanix601** [comment]
绑定模式手动指定TwoWay
`Visibility="{Binding WindowVisibility,Converter={local:BoolToVisibilityConvert}, Mode=TwoWay}"`

**Reply by 158545614** [reply]
大佬 默认是oneway吗

**Reply by arcanix601** [reply]
是的
![image](https://github.com/user-attachments/assets/e4a1dd04-3403-4731-9170-b919f1891c38)

**Reply by 158545614** [reply]
谢谢大佬指点！

**Comment by TRARS** [comment]
要操作大写开头的WindowVisibility

**Reply by 158545614** [reply]
谢谢大佬，我操作的就是大写的WindowVisibility
